core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -98,14 +105,22 @@ class Replica(val brokerId: Int, val topicPartition: TopicPartition) extends Log
    * fetch request is always smaller than the leader's LEO, which can happen if small produce requests are received at
    * high frequency.
    */
-  def updateFetchState(
+  def updateFetchStateOrThrow(
     followerFetchOffsetMetadata: LogOffsetMetadata,
     followerStartOffset: Long,
     followerFetchTimeMs: Long,
     leaderEndOffset: Long,
     brokerEpoch: Long
   ): Unit = {
     replicaState.updateAndGet { currentReplicaState =>
+      val cachedBrokerEpoch = if (metadataCache.isInstanceOf[KRaftMetadataCache])
```

Comment:
nit: in scala, it's usually cleaner to use a match instead of `isInstanceOf`.

Reply:
Thanks!